src/Datadog.Trace.ClrProfiler.Managed/Instrumentation.cs

```diff
@@ -126,5 +130,37 @@ private static void StartDiagnosticManager()
             DiagnosticManager.Instance = diagnosticManager;
         }
 #endif
+
+        private static void TryLoadVendorPlugin()
+        {
+            string pluginName = GlobalSettings.Source.VendorPluginName;
+            if (!string.IsNullOrWhiteSpace(pluginName))
+            {
+                string pluginPath = Path.Combine(
+                    Path.GetDirectoryName(typeof(Instrumentation).Assembly.Location),
+                    pluginName);
+
+                if (File.Exists(pluginPath))
+                {
+                    try
+                    {
+                        Assembly pluginAssembly = Assembly.LoadFrom(pluginPath);
+
+                        GlobalSettings.SetVendorPlugin(pluginAssembly);
+
+                        Log.Information("Vendor plugin assembly loaded '{0}'.", pluginAssembly.FullName);
+                    }
+                    catch (Exception ex)
```

Comment:
do not catch general exceptions like that. What if we catch a runtime exception like `OutOfMemoryException`? You can check all "normal" exceptions which `Assembly.LoadFrom` can throw https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.loadfrom?view=net-5.0 (check in all supported frameworks).

Reply:
Added prototype to handle different exceptions. Currently can't find any existing pattern, lot of places are catching general exception.